src/extension.ts

```diff
@@ -62,9 +63,16 @@ export async function activate(context: vscode.ExtensionContext) {
 		}
 	});
 
-	// set up our completion provider
+	// set up our completion providers
 	let markdown:vscode.DocumentSelector = { scheme: 'file', language: 'markdown', pattern: '**/**.didact.md' };
-	vscode.languages.registerCompletionItemProvider(markdown, new DidactUriCompletionItemProvider());
+	let markdownProvider = new DidactUriCompletionItemProviderMarkdown();
+	markdownProvider.setContext(context);
```

Comment:
what about providing the context as a aprameter of the constructor? it will avoid to have an extra line and isn't the context mandatory? it will avoid thatif we want to reuse that we forgot to set the context.